app/hotels/src/navigation/AllHotelsNavigationScreen.js

```diff
@@ -87,7 +89,19 @@ class AllHotelsNavigationScreen extends React.Component<Props> {
     };
   };
 
-  openSingleHotel = searchParams =>
+  componentDidMount = () => {
+    if (Platform.OS === 'android' && this.backButtonListner === null) {
+      this.backButtonListner = BackHandler.addEventListener(
+        'hardwareBackPress',
+        () => {
+          this.props.onBackClicked();
+          return true;
+        },
+      );
+    }
+  };
```

Comment:
Just out of curiosity - why does it work in our development environment? It looks like you are workarounding something you should not. Do they extend `ReactActivity`?

Reply:
Yes, in their activity they have to define
```
@Override
    public void onBackPressed() {

        if (mReactInstanceManager != null) {
            mReactInstanceManager.onBackPressed();
        } else {
            super.onBackPressed();
        }
    }
```

Which basically lets react-native handle navigation. When back button is pressed in the root view, react-native has no more views to go back to. Nothing happens when clicking back button, so we have to trigger the onBackClick callback.

Reply:
Which means that clicking the android backbutton leads back to this
```
onBackClicked = () => {
    if (NativeModules.RNNavigationModule) {
      NativeModules.RNNavigationModule.leaveHotels();
    } else {
      this.props.onBackClicked();
    }
  };
```
which is why it works in development